TraitsExpanded/TraitsExpandedSubModule.cs

```diff
@@ -1,16 +1,17 @@
 ﻿using TaleWorlds.Core;
 using TaleWorlds.MountAndBlade;
 using TaleWorlds.CampaignSystem;
+using TraitsExpanded.TraitSets;
 
 namespace TraitsExpanded
 {
     public class TraitsExpandedSubModule : MBSubModuleBase
     {
-	    private readonly TraitSystemBehavior traitSystemBehavior = new TraitSystemBehavior();
+	    private TraitSystemBehavior traitSystemBehavior;
 
 	    protected override void OnApplicationTick(float dt)
 	    {
-		    traitSystemBehavior.Tick();
+		    traitSystemBehavior?.Tick();
```

Comment:
in general I think we should crisp about what's allowed to be null and when. for stuff that we control, which should never be null, we shouldn't add null checks for it - if it's null, it's a coding bug. for things the game controls, we should figure out if there's a consistent way we can avoid e.g. listening for `OnApplicationTick` if `AddBehaviors` hasn't been called

Reply:
I second that, I like robust programming and catching and handling all that can go wrong, but if we can already avoid passing invalid values on an earlier stage in the process, we don't need to do that everywhere.

The question in this case should be: Is a tick possible where no traitSystemBehavior is initialized? If so, does this indicate something went wrong and should i probably try to initialize it now?

Reply:
I agree, but this will require a lot of testing to see which events actually fire when. But this is definitely something that we should keep an eye on